### PR TITLE
Payment flow foundation

### PR DESCRIPTION
**Pull request by derekdowling**
Okay this provides the most basic payment flow with proper notifications that hooks into the Negotiation View.
* Buyer Start Purchase
* Buyer Confirm
* Seller Approve
All using the realtime update system. Actual charging is disabled for now just to get the flow going but all of the logic is almost ready to go. I'd rather merge this now than later when it's impossible because there is so much stuff.

### CODE REVIEW

File: TextCrunch/NegotiationViewController.swift [hunk 1]
```diff
@@ -135,12 +150,49 @@ class NegotiationViewController : UIViewController, UITableViewDataSource, UITab
...
+                resultNegotiation?.purchaseRequested == true &&
+                userIsSeller == true
+            )
+            {
+                self.confirmSale()
+                
+            // if the seller confirms, pay up and complete from the user side
+            // because we need their PayPal metadata
+            } else if (
+                self.negotiation.sellerConfirmation == false &&
```
Comment (gcoomber) [inline TextCrunch/NegotiationViewController.swift]: Should be same line as else for consistency.

File: TxtCrunchCloud/cloud/base64.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Making up for a lack of Btoa in Cloud Code
+var keyStr = 'ABCDEFGHIJKLMNOP' +
```
Comment (gcoomber) [inline TxtCrunchCloud/cloud/base64.js]: I'm probably not the best one to review js code, but I dont see any red flags.

File: TextCrunch/PaymentManager.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        ]
+        
+        var response = PFCloud.callFunction("charge", withParameters: data) as [String:AnyObject]
+        print(response)
+        return response["success"] as Bool
+    }
+    
+    class func capturePayment(negotiation: Negotiation) -> Bool {
+
+//        var data : NSDictionary = {
```
Comment (gcoomber) [inline TextCrunch/PaymentManager.swift]: Is this a TODO or should this stuff be removed?

Comment (derekdowling) [inline TextCrunch/PaymentManager.swift]: That's all TODO still. This code is all good. Just needs a bit of bug hunting on the cloud side still.

**Pull request merged**